code/modules/modular_computers/laptop_vendor.dm

```diff
@@ -277,20 +306,20 @@ obj/machinery/lapvend/attackby(obj/item/weapon/W as obj, mob/user as mob)
 	var/datum/money_account/customer_account = get_account(I.associated_account_number)
 	if (!customer_account || customer_account.suspended)
 		ping("Connection error. Unable to connect to account.")
-		return FALSE
+		return 0
 
 	if(customer_account.security_level != 0) //If card requires pin authentication (ie seclevel 1 or 2)
 		var/attempt_pin = input("Enter pin code", "Vendor transaction") as num
 		customer_account = attempt_account_access(I.associated_account_number, attempt_pin, 2)
 
 		if(!customer_account)
 			ping("Unable to access account: incorrect credentials.")
-			return FALSE
+			return 0
 
 	if(total_price > customer_account.money)
 		ping("Insufficient funds in account.")
-		return FALSE
+		return 0
 	else
 		transaction_template.set_amount(-total_price)
 		transaction_template.purpose = "Purchase of [(devtype == 1) ? "laptop computer" : "tablet microcomputer"]."
-		return transaction_template.apply_to(customer_account)
+		return 1
```

Comment:
Why do you remove payment itself?
![11111](https://user-images.githubusercontent.com/2458675/47558511-be579780-d92c-11e8-92b9-0f268bdd57c8.png)